splinterd/src/config/builder.rs

```diff
@@ -12,19 +12,229 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-use crate::config::PartialConfig;
+use std::path::Path;
+
+use crate::config::error::ConfigError;
+use crate::config::{Config, PartialConfig};
 
 pub trait PartialConfigBuilder {
     /// Takes all values set in a config object to create a PartialConfig object.
     ///
     fn build(self) -> PartialConfig;
 }
 
+fn get_file_path(cert_dir: &str, file: &str) -> String {
```

Comment:
At least before, if ca_certs, client/server_cert, client/server_keys they took a full path, and would not be prepended with the cert_dir. 

For example if:
`--server-key ../certs/server.key` was passed that location would be used. 

Now it searches this location `/etc/splinter/certs/../certs/server.key` instead.

Reply:
Addressed this by checking the source for the certificate file path in the main `build` function. Now, if the certificate file comes from a config (besides default), the value is passed directly like that. However, if the certificate file is from the default config, the cert_dir will be appended to it.